MCR-2424 MCRXMLMetadataManager Interface still carries Store-specific classes

OPNA2608 opened the pull request:
[Link to jira](https://mycore.atlassian.net/browse/MCR-2424).

OPNA2608 committed:
MCR-2424 MCRXMLMetadataManager Interface still carries Store-specific classes

Remove return values from create & update methods

OPNA2608 committed:
MCR-2424 MCRXMLMetadataManager Interface still carries Store-specific classes

Make getVersionedMetaData private

OPNA2608 commented on mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestObjects.java:
```diff
@@ -374,10 +374,9 @@ public Response getObjectVersions(@Parameter(example = "mir_mods_00004711") @Pat
+        List<Long> versions = MCRXMLMetadataManager.instance().listRevisions(id);
         return Response.ok()
-            .entity(new GenericEntity<>(versions, TypeUtils.parameterize(List.class, MCRMetadataVersion.class)))
+            .entity(new GenericEntity<>(versions, TypeUtils.parameterize(List.class, Long.class)))
```
@rsteph-de I was told by Jens to ping you about this because it affects the REST API.

MyCoRe-Travis commented:
Something gone wrong on 4478624877afb87ac2123c264ae555dc8e5fa0c9! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7247-pull_request-1262

yagee-de commented:
Just some random thought here:
Long may not fit for all version systems in the future (e.g. git based) but is enough for SVN and OCFL
Just keeping a "long" around and asking for revision information (author, date, type) may result in a lot of I/O that would come on-top.
Maybe we could use a generic way to describe a revision and have it's (serializable) id be implemented by the store. long, sha1 ...

sebhofmann commented:
Please rebase to run new github actions

OPNA2608 committed:
MCR-2424 MCRXMLMetadataManager Interface still carries Store-specific classes

Change requests:
- `long` revision type -> something more generic
  I've changed the returned revision type to a String and convert back to a `Long` where needed.
  Fully converting everything to properly handle String revisions is gonna require alot more work.
- Generic version type to cut down on I/O
  I've created an `abstract MCRAbstractMetadataVersion<T>` type and made the current `MCRMetadataVersion` an extension
  of it.

rsteph-de reviewed (approved):
looks good for me
